examples/arduino/arduino.ino

```diff
@@ -0,0 +1,116 @@
+#endif
+
+// 5 second timeout for retrying the hub.set request.
+#define HUB_SET_RETRY_SECONDS 5
+// Fetch every 20 seconds.
+#define FETCH_INTERVAL_MS (20 * 1000)
+
+// A struct to cache the values of environment variables.
+typedef struct {
+    char valueA[16];
+    char valueB[16];
+    char valueC[16];
+} EnvVarCache;
+
+Notecard notecard;
+NotecardEnvVarManager *envVarManager = NULL;
+EnvVarCache envVarCache;
+uint32_t lastFetchMs = 0;
+
+void envVarManagerCb(const char *var, const char *val, void *userCtx)
+{
+    EnvVarCache *cache = (EnvVarCache *)userCtx;
+
+    // Cache the values for each variable.
+    if (strcmp(var, "variable_a") == 0) {
+        strncpy(cache->valueA, val, sizeof(cache->valueA));
+    }
+    else if (strcmp(var, "variable_b") == 0) {
+        strncpy(cache->valueB, val, sizeof(cache->valueA));
+    }
+    else if (strcmp(var, "variable_c") == 0) {
+        strncpy(cache->valueC, val, sizeof(cache->valueA));
+    }
+}
+
+// These are the environment variables we'll be fetching from the Notecard.
+const char *envVars[] = {
+    "variable_a",
+    "variable_b",
+    "variable_c"
+};
+static const size_t numEnvVars = sizeof(envVars) / sizeof(envVars[0]);
+
+void setup()
+{
+    delay(2500);
+    Serial.begin(115200);
+    // Set up debug output via serial connection.
+    notecard.setDebugOutputStream(Serial);
+
+    // Initialize the physical I/O channel to the Notecard.
+    Wire.begin();
+    notecard.begin();
+
+    // Issue the hub.set request to set the ProductUID on the Notecard.
+    J *req = notecard.newRequest("hub.set");
+    if (PRODUCT_UID[0]) {
+       JAddStringToObject(req, "product", PRODUCT_UID);
+    }
+    JAddStringToObject(req, "mode", "continuous");
+    JAddBoolToObject(req, "sync", true);
+    // Send the request with a retry timeout. If the Notecard has just started
+    // up, it may need a moment before it's able to receive and respond to
+    // requests.
+    if (!notecard.sendRequestWithRetry(req, HUB_SET_RETRY_SECONDS)) {
+        notecard.logDebug("hub.set request failed.\r\n");
+        return;
+    }
+
+    // Allocate the environment variable manager.
+    envVarManager = NotecardEnvVarManager_alloc();
+    if (envVarManager == NULL) {
+        notecard.logDebug("Failed to allocate env var manager.\r\n");
+        return;
+    }
+
+    // Set the callback for the manager, and give it a pointer to our cache
+    // so that we can store the values of the environment variables we're
+    // fetching.
+    if (NotecardEnvVarManager_setEnvVarCb(envVarManager, envVarManagerCb,
+        &envVarCache) != NEVM_SUCCESS) {
+        notecard.logDebug("Failed to set env var manager callback.\r\n");
+        return;
+    }
+}
+
+void loop()
+{
+    // Fetch the environment variables every FETCH_INTERVAL_MS milliseconds.
+    uint32_t currentMs = NoteGetMs();
```

Comment:
`NoteGetMs()` is an abstraction that we typically use internally for libraries that may not be written to the Arduino API. Personally, I would use `millis()` here, but this is harmless.

```suggestion
    uint32_t currentMs = millis();
```